src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -933,10 +934,35 @@ FixedwingPositionControl::control_position(const math::Vector<2> &curr_pos, cons
 			_att_sp.yaw_body = 0;
 		}
 
+	} else if (_control_mode.flag_control_altitude_enabled
+		   && _control_mode.flag_control_offboard_enabled
+		   && pos_sp_curr.valid) {
```

Comment:
Why is a valid position setpoint needed here? Isn't altitude mode intended to hold whatever altitude one is currently at? (and then ascend/descend by changing the setpoint via sink/climb-rate * delta_time ?).

Reply:
As far as I'm aware this flag encompasses both `yawspeed` and as well as the `position`/`altitude`. In regards to the idea of only using climb/sink rate to change altitude, that does seem like the more generic approach, I'll change it, thanks!